platform/server/http/HttpServer.ts

```diff
@@ -1,42 +1,89 @@
 import * as express from 'express';
-import * as bodyParser from 'body-parser';
 import * as http from 'http';
+import * as https from 'https';
+import { promisify } from 'util';
+import { readFileSync } from 'fs';
+
+import { HttpConfig } from './HttpConfig';
+import { Env } from '../../config';
+import { Logger } from '../../logging';
 import { Router } from './Router';
 
 export class HttpServer {
   private readonly app: express.Application;
-  private readonly httpServer: http.Server;
+  private server: http.Server | https.Server | null = null;
 
-  constructor() {
+  constructor(private readonly log: Logger, private readonly env: Env) {
     this.app = express();
-    this.app.use([
+
+    // We will register body parser only for routes defined in the new platform.
+    /*this.app.use([
       bodyParser.json(),
+      bodyParser.raw({ type: 'application/x-ndjson' }),
       bodyParser.urlencoded({ extended: false })
-    ]);
-    this.httpServer = http.createServer(this.app);
+    ]);*/
   }
 
   isListening() {
-    return this.httpServer.listening;
+    return this.server && this.server.listening;
   }
 
   registerRouter(router: Router<any>) {
     this.app.use(router.path, router.router);
   }
 
-  start(port: number, host: string) {
-    return new Promise((resolve, reject) => {
-      this.httpServer.listen(port, host, (err?: Error) => {
-        if (err != null) {
-          reject(err);
-        } else {
-          resolve();
-        }
-      });
-    });
+  async start(config: HttpConfig) {
+    this.server = this.initializeServer(config);
+
+    let port = config.port;
+
+    const proxy = this.env.getProxy();
+    if (proxy) {
+      proxy.bind(this.server);
+      port = proxy.getPort();
```

Comment:
The real port is passed by old Kibana in `setup_connection -> proxy.listen(...)` so we don't have to think which port to use here, just grab what old platform gave us.